src/release.ss

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env gxi-script
+;; -*- Gerbil -*-
+;;; Release script for Gerbil.
+;; Usage: ./release.ss [version]
+;;
+;; Use it from a fully committed git tree.
+;;
+;; If a release version is specified (contains no "-"), a new release commit is made and tagged.
+;; If a development version is specified, a new development commit is made and tagged.
+;;
+;; If no version is specified, one is autodetected:
+;; If the current commit is a development commit, a release commit will be made and tagged
+;; If the current commit is a release commit, a development commit will be made and tagged
+;; e.g. v0.12-DEV-666-g1234567 will be turned into v0.13
+;; and v0.13 will be turned into v0.13-DEV
+
+(import
+  :gerbil/gambit/ports
+  :std/format :std/misc/ports :std/srfi/13 :std/srfi/14 :std/sugar :std/pregexp)
+
+;; Where the root to gerbil is.
+(def gerbil-root
+  (path-normalize (path-expand ".." (path-directory (this-source-file)))))
+
+;; Change the directory to the root of gerbil.
+(current-directory gerbil-root)
+
+;; Check that .git is there indeed
+(or (file-exists? ".git") (error "Cannot find .git directory"))
+
+;; Path to the version file
+(def version-path
+  "src/gerbil/runtime/gx-version.scm")
+
+;; Path to the .gitignore file
+(def ignore-path
+  ".gitignore")
+
+;; Error
+(def (raise-process-error status settings)
+  (error "Process exited with non-zero status" status settings))
+
+(def (check-process-error process settings)
+  (let ((status (process-status process)))
+    (unless (zero? status)
+      (raise-process-error status settings))))
+
+;; Get the output of a process, by default as a string.
+;; The caller may specify via the reader: argument an alternate function taking an input-port;
+;; the value returned will be that of the function.
+;; If the check-status?: argument is true, then the process exit status will be checked,
+;; and an error will be raised if it isn't 0.
+;; If the check-status?: argument is false,
+;; the process exit status will be returned as a second value.
+(def (process-output
+      settings
+      reader: (reader read-all-as-string)
+      check-status?: (check-status? #t))
+  (let ((process (open-process
+                   `(,@settings
+                     show-console: #f
+                     stdin-redirection: #t
+                     stdout-redirection: #t
+                     stderr-redirection: #f))))
+    (try
+      (close-output-port process)
+      (let ((output (reader process)))
+        (if check-status?
+          (begin
+            (check-process-error process settings)
+            output)
+          (values output (process-status process))))
+      (finally
+       (close-port process)
+       (process-status process)))))
+
+;; Just run the process on the current console
+(def (run-git . arguments)
+  (let ((settings [path: "git" arguments: arguments stdout-redirection: #f]))
```

Comment:
here for reference: `stdout-redirection:` is set twice in the process settings; which one wins?

Reply:
cf above, it's last wins so we end up with stdout-redirection regardless.

Reply:
Thanks, fixing.